### DISCUSSION TITLE
How to calculate the Jacobian in [ComputeSimoHughesJ2PlasticityStress]

### DISCUSSION THREAD

**Discussion by rh201** [idaholab/moose#24302] [Q&A Modules: Solid mechanics] [answered]
Hallo MOOSEers,

I'm trying to use [ComputeSimoHughesJ2PlasticityStress] module, as listed here [https://mooseframework.inl.gov/source/materials/lagrangian/ComputeSimoHughesJ2PlasticityStress.html](url). When I check the source code of it, I cannot figure out how does the jacobian matrix is calculated, specifically the derivatives inside, like 'd_tau_d_F', '_d_be_d_F', '_d_deltaep_d_betr', '_d_n_d_be' , '_d_R_d_betr', '_d_J_d_betr'. I checked to the reference of it (Simo and Hughes (2006), Borden et al. (2016).), but I didn't find these derivatives. So I'm writing to ask how these values are derived, and it would be great if you could refer me to any literatures. Thanks!

**Comment by hugary1995** [comment, marked as answer]
You can update the Jacobian iteratively (the one I implemented) like this: 
[SimoHughesJ2Jacobian.pdf](https://github.com/idaholab/moose/files/11426128/SimoHughesJ2Jacobian.pdf)

or use the implicit function theorem, which is easier in my opinion.

I haven't had a chance to publish the above derivation, though it's pretty straightforward IMO.

**Reply by rh201** [reply]
Thank you! @hugary1995
